cmd/release-controller/sync_verify_prow.go

```diff
@@ -21,13 +21,39 @@ import (
 func (c *Controller) ensureProwJobForReleaseTag(release *Release, verifyName string, verifyType ReleaseVerification, releaseTag *imagev1.TagReference) (*unstructured.Unstructured, error) {
 	jobName := verifyType.ProwJob.Name
 	prowJobName := fmt.Sprintf("%s-%s", releaseTag.Name, verifyName)
-	obj, exists, err := c.prowLister.GetByKey(fmt.Sprintf("%s/%s", c.prowNamespace, prowJobName))
-	if err != nil {
-		return nil, err
-	}
-	if exists {
-		// TODO: check metadata on object
-		return obj.(*unstructured.Unstructured), nil
+
+	if verifyType.Optional || releaseVerificationBlockingJobAttempts <= 1 {
+		obj, exists, err := c.prowLister.GetByKey(fmt.Sprintf("%s/%s", c.prowNamespace, prowJobName))
+		if err != nil {
+			return nil, err
+		}
+		if exists {
+			// TODO: check metadata on object
+			return obj.(*unstructured.Unstructured), nil
+		}
+	} else {
+		jobAttempts := 0
+		for _, job := range c.prowLister.List() {
```

Comment:
This is incredibly expensive - we probably can't afford to do this.  Note that also we clean up jobs regularly, so this wouldn't get at jobs run a day or two ago.

The count needs to be tracked with the job info and the invariant imposed there.  The stored data about verification in the annotation would have to change to accept multiple runs, not just a single run.